app/src/main/java/org/oppia/android/app/topic/revisioncard/RevisionCardViewModel.kt

```diff
@@ -114,4 +129,31 @@ class RevisionCardViewModel @Inject constructor(
       is AsyncResult.Success -> revisionCardResult.value
     }
   }
+
+  class Factory @Inject constructor(
+    private val activity: AppCompatActivity,
+    private val topicController: TopicController,
+    private val oppiaLogger: OppiaLogger,
+    @TopicHtmlParserEntityType private val entityType: String,
+    private val translationController: TranslationController
+  ) {
+    fun create(
```

Comment:
This & all methods above, and for the Factory class: we need KDocs here.

Reply:
done -- PTAL.

Reply:
All other public fields & methods need KDocs, too.

Reply:
To clarify, please document all non-private members and all properties in the view model's constructor.

Reply:
Please make sure to read these comments, fully. You didn't document the properties or the fields, only the methods.